framework/src/main/java/io/github/kgress/scaffold/webdriver/BasePage.java

```diff
@@ -18,13 +24,73 @@
 public abstract class BasePage {
 
     /**
+     * {@link Deprecated} in favor of {@link #verifyIsOnPage(AbstractWebElement...)}. This method will be phased out in the next
+     * breaking release.
+     *
      * A method to be overridden by the implementing project. Typically, it's best to use elements that are unique to
      * the page object that is being navigated to.
      *
      * @return the {@link Boolean} value to determine if the page is correctly loaded
      */
+    @Deprecated
     public abstract boolean isOnPage();
 
+    /**
+     * A method that is used to verify a number of elements are displayed on the page when navigating to it. In addition,
+     * it will check to ensure the DOM is in a ready and loaded state prior to proceeding. This will help alleviate
+     * issues with websites where it takes additional time to load the page due to extraneous db calls or slow loading JS.
+     *
+     * It's best to use this method in the constructor of the page object as it will be invoked at the time the page is
+     * instantiated. For example:
+     *
+     * <pre>{@code
+     * @Getter
+     * public class LoginPage() {
+     *     private InputWebElement emailInput = new InputWebElement("#email");
+     *     private InputWebElement passwordInput = new InputWebElement("#password");
+     *
+     *     public LoginPage() {
+     *         isOnPage(getEmailInput, getPasswordInput);
+     *     }
+     * }
+     * }</pre>
+     *
+     * The elements passed in to this method will be checked by selenium to see if they are displayed. Make sure to pass in
+     * elements that are unique to the page that won't show up on other pages. For example, a login page will have an email
+     * and password input and would pass in those elements as parameters. Don't use elements from headers or a logo that
+     * might appear on all of the pages.
+     *
+     * @param element the element(s) that will be checked if displayed
+     * @return the {@link Boolean} value to determine if the page is correctly loaded
+     */
+    public Boolean verifyIsOnPage(AbstractWebElement... element) {
+        // Get the list of elements and iterate over them so we can remove any duplicates
+        var listOfElements = Arrays.stream(element)
+                .distinct() //removes duplicates
```

Comment:
I'm not 100% sure of this, but I don't think this is going to behave the way you want it to behave, since `AbstractWebElement` doesn't appear to have overridden `equals()` and `hashCode()`. I didn't see a unit test that verifies that a duplicate set of elements will be collapsed down to 1 in the `distinct()` call, but I believe as it is, it's impossible for `distinct()` to actually remove elements from the list.  I believe `distinct()` relies on `equals()` for comparison, which by default will compare memory address, not "sameness". Since it's impossible for a memory address to be the same between two otherwise identical elements, I don't think it will pull a duplicate from the list.

If I had more time, I'd add a unit test that verifies this, or I'd look up the javadocs for `stream().distinct()`. That's what I'd recommend, just to make sure you don't get any wonky or surprising behavior from this method.

Reply:
I've done some end to end testing with this using the heb ui automation, and I've found that passing in duplicate elements do filter down to unique instance of the element.

In the example I showed earlier, I passed in three email input web elements and it was filtered down to one. I believe it's comparing the hashcode for the removal. I don't know all of the inner workings for streams but I'm pretty sure this will do the trick for us.

As far as unit testing, I'm not 100% sure that testing stream functionality is worthwhile.

Reply:
After some additional research, I think we are still okay here. `stream.distinct()` will use the `equals` implementation on the object its doing the comparison on. Since `AbstractWebElement` doesn't have one, it defaults to `Object`. In this case, since we're passing in the same element every time, and not a new instance of that same element, the elements are considered duplicate because they use the same ID. 

Furthermore, even if users create new instances of the same element, it will not break the method. The trim down was added as a curtesy to speed up the process if a user was passing in duplicate elements.

Example of a use case where duplicates won't be trimmed:
`verifyIsOnPage(new InputWebElement("#email), new InputWebElement("#email), new InputWebElement("#email));`

Example of the how the method is intended where duplicates will be trimmed:
`verifyIsOnPage(getEmailInput, getEmailInput, getEmailInput);`

I already have it documented on the method how to use it. I think we'll be fine in this case!

Reply:
That's great--exactly answers my question, and I'm sorry I didn't take the time to do that research myself. Everything else looks good to me. I haven't run any code through it, but I didn't have any other concerns.

Reply:
Awesome! :D